### PR TITLE
Updating deepchem's splitters tutorial

### PR DESCRIPTION
**Pull request by Dragonwagon18**
Divided Splitters into Two Groups:
General Splitters
Molecular Splitters

Added Descriptions on How the Splitters Work:
General Splitters: These are applicable across different domains and datasets.
Molecular Splitters: Specifically designed for chemical datasets and molecular structures.

Added Fingerprint Splitter Implementation:
Integrated the fingerprint splitter into the code for splitting and training the BACE dataset.

Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

Checklist
- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by GreatRSingh** [comment]
LGTM

cc @rbharath

**Review by GreatRSingh — APPROVED** [review]
LGTM

cc: @rbharath

File: examples/tutorials/Working_With_Splitters.ipynb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
```
Comment (shreyasvinaya) [inline examples/tutorials/Working_With_Splitters.ipynb:1]: <p>Could you resolve the errors at the top and push the notebook again</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3837/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (shreyasvinaya) [inline examples/tutorials/Working_With_Splitters.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #6.        tasks, datasets, transformers = dc.molnet.load_bace_classification(featurizer='ECFP', splitter=splitter)
</pre><p>any specific reason we are switching from tox21 dataset to BACE?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3837/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (rbharath) [inline examples/tutorials/Working_With_Splitters.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.     author={Eastman, Peter and Ramsundar, Bharath}, 
</pre><p>Add yourself as an author, between me and Peter</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/3837/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Review by rbharath — COMMENTED** [review]
LGTM to merge once you've added yourself as an author

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**